de-lib/src/main/java/org/iplantc/de/apps/client/presenter/list/AppsListPresenterImpl.java

```diff
@@ -277,6 +281,7 @@ protected void getAppsWithSelectedCategory() {
     public void onOntologyHierarchySelectionChanged(OntologyHierarchySelectionChangedEvent event) {
         listView.setLoadingMask(true);
         selectedHierarchy = event.getSelectedHierarchy();
+        appCategory = null;
```

Comment:
resetting `selectedHierarchy` when `appCategory` is selected and vice-versa fixes refresh logic in most cases. I think there is still issue when your search and refresh...